src/components/multi_select_state.js

```diff
@@ -68,11 +68,15 @@ const withMultiSelectState = WrappedComponent =>
     getNewSelectedItems(itemId) {
       const { items } = this.props;
       const { selectedItems } = this.state;
-      return items.filter(
+      const itemsInSource = items.filter(
```

Comment:
A bit odd having a "in" in one parameter and "from" in another.
How about sourceItems and destinationItems?